### PR TITLE
Provide gPRC data transport protocol define

### PR DESCRIPTION
**Pull request by mrproliu**
Follow [main issue](https://github.com/apache/skywalking/issues/4808), provide the data transport protocol defined.

### CODE REVIEW

File: language-agent/Meter.proto [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.apache.skywalking.apm.network.language.agent.v3";
+
+import "common/Common.proto";
+
+service MeterReportService {
+    rpc collect (stream MeterData) returns (Commands) {
```
Comment (wu-sheng) [inline language-agent/Meter.proto:27]:
```suggestion
    // Meter data is reported in a certain period. The agent/SDK should report all collected metrics in this period through one stream.
    rpc collect (stream MeterData) returns (Commands) {
```

File: language-agent/Meter.proto [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+}
+
+// Label in the meter
```
Comment (wu-sheng) [inline language-agent/Meter.proto:31]:
```suggestion
// Label of the meter
```

File: language-agent/Meter.proto [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+message Label {
+    string name = 1;
+    string value = 2;
+}
+
+// Work for histogram, defined the bucket
```
Comment (wu-sheng) [inline language-agent/Meter.proto:37]:
```suggestion
// The histogram definition.
```

File: language-agent/Meter.proto [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+message MeterBucketValue {
+    int32 bucket = 1;
```
Comment (wu-sheng) [inline language-agent/Meter.proto:39]:
```suggestion
    // The bucket is [0, $value]
    int32 bucket = 1;
```

Is this explanation right? Are we using the Prometheus style too? I remember we said we have min/max values boundaries. Do I miss anything?

Comment (mrproliu) [reply language-agent/Meter.proto:39]: We will have boundaries settings in the `API` level, `exceptMinValue` and `exceptMinValue` will work on it. But at the default, it also is `Integer.MIN_VALUE` and `Integer.MAX_VALUE`. 
Using Prometheus style to relate the bucket and value. such as [previous bucket, $value].

Comment (wu-sheng) [reply language-agent/Meter.proto:40]: I think the boundary should be very clear at the protocol level. MIN and MAX seem not a good way. API is another thing, let's talk later.

> Using Prometheus style to relate the bucket and value. such as [previous bucket, $value].

I am not asking to use it or not. I am asking about what is your design?

**Comment by mrproliu** [comment]
> Does this proto file have been tested by the grpc tool? If so, I could focus on the logic only.

Has tested. The generate code is correct.

File: language-agent/Meter.proto (hunk 4 above)
Comment (wu-sheng) [reply language-agent/Meter.proto:40]: I updated the comments based on our offline discussion. Take a look.

File: language-agent/Meter.proto [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    }
+}
+
+// Label of the meter
+message Label {
+    string name = 1;
+    string value = 2;
+}
+
+// The histogram definition
```
Comment (wu-sheng) [inline language-agent/Meter.proto:38]:
```suggestion
// The histogram element definition. It includes the bucket lower boundary and the count in the bucket.
```

File: language-agent/Meter.proto [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    string name = 1;
+    // labels
+    repeated Label labels = 2;
+    // customize the buckets
+    repeated MeterBucketValue values = 3;
+}
+
+// single meter data, if the same metrics have a different label, they will separate.
+message MeterData {
+    // only transform single one
```
Comment (wu-sheng) [inline language-agent/Meter.proto:66]:
```suggestion
    // Meter data could be a single value or histogram.
```

File: language-agent/Meter.proto [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    oneof metric {
+        MeterSingleValue singleValue = 1;
+        MeterHistogram histogram = 2;
+    }
+    // current service name, set value when fist collect stream data
```
Comment (wu-sheng) [inline language-agent/Meter.proto:71]:
```suggestion
    // Service name, be set value in the first element in the stream-call.
```

File: language-agent/Meter.proto [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    string service = 3;
+    // current service instance name, set value when fist collect stream data
```
Comment (wu-sheng) [inline language-agent/Meter.proto:73]:
```suggestion
    // Service instance name, be set value in the first element in the stream-call.
```

File: language-agent/Meter.proto [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    string serviceInstance = 4;
+    // send meter timestamp, set value when fist collect stream data
```
Comment (wu-sheng) [inline language-agent/Meter.proto:75]:
```suggestion
    // Meter data report time, be set value in the first element in the stream-call.
```

File: language-agent/Meter.proto (hunk 4 above)
Comment (mrproliu) [reply language-agent/Meter.proto:40]: Thanks for helping me to fix the document.

**Pull request merged**